corehq/motech/repeaters/repeater_generators.py

```diff
@@ -261,6 +266,103 @@ def get_test_payload(self, domain):
         )
 
 
+@attr.s
+class CaseTypeReferralConfig(object):
+    #if false, listed_properties is a whitelist
+    use_blacklist = attr.ib()
+    listed_properties = attr.ib()
+    constant_properties = attr.ib()
+
+
+class ReferCasePayloadGenerator(BasePayloadGenerator):
+
+    def get_payload(self, repeat_record, payload_doc):
+
+        def _update_case_id(original_case_id, case_id_map):
+            if original_case_id in case_id_map:
+                new_case_id = case_id_map[original_case_id]
+            else:
+                new_case_id = uuid4().hex
```

Comment:
Could this instead call a utility function for generating a new case id? It's been mildly annoying in the past that we have multiple ways of generating xform namespaces, it'd be nice to be more consistent with case ids.